### PR TITLE
adding debug logging option and some debug lines for request body

### PR DESCRIPTION
**Pull request by danielwhatmuff**
- Adding debug option
- Switching levels for error logs
- Adding listen port and host to startup log for better visibility

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -167,6 +171,16 @@ func main() {
 				Usage:  "the path to the file containing the private key pem `PATH`",
 				EnvVar: "TLS_KEY",
 			},
+			cli.BoolFlag{
+				Name:   "debug",
+				Usage:  "enable debug logging",
+				EnvVar: "DEBUG",
+			},
+			cli.StringFlag{
```
Comment (gambol99) [inline main.go]: change to the `cli.IntFlag`  and add a `Destination: &defaultWatchTimeout` (needs changing to a var not a const though)

File: go/api_default.go [hunk 2]
```diff
@@ -94,8 +109,12 @@ func handleNotFoundError(w http.ResponseWriter, err error) {
...
 }
 
-func waitForServiceAccountSecret(timeSecs int, namespace, serviceAccountName string, clientset *kubernetes.Clientset) (serviceAccount *apicorev1.ServiceAccount, found bool, err error) {
-	for i := 0; i < timeSecs; i++ {
+func waitForServiceAccountSecret(namespace, serviceAccountName string, clientset *kubernetes.Clientset) (serviceAccount *apicorev1.ServiceAccount, found bool, err error) {
+	timeoutSecs, err := getEnv("TIMEOUT", 5)
+	if err != nil {
+		logrus.Errorln(err)
+	}
+	for i := 0; i < timeoutSecs; i++ {
```
Comment (gambol99) [inline go/api_default.go]: this is more a max retry than timeout though ..

File: main.go (hunk 1 above)
Comment (danielwhatmuff) [reply main.go]: Oh cool. I wondered how to pass it from there.

File: go/api_default.go [hunk 3]
```diff
@@ -31,6 +33,19 @@ import (
 
 var clientset *kubernetes.Clientset
 
+const defaultWatchTimeout int = 5
+
+func getEnv(key string, fallback int) (value int, err error) {
```
[the remark this answers is not in the document]
Comment (danielwhatmuff) [reply go/api_default.go]: Removed

File: main.go (hunk 1 above)
Comment (danielwhatmuff) [reply main.go]: Added but getting a nil pointer when referencing in `go/api_default.go`

File: go/api_default.go (hunk 2 above)
Comment (danielwhatmuff) [reply go/api_default.go]: Renamed.

File: go/api_default.go [hunk 4]
```diff
@@ -60,14 +46,13 @@ func getClient(server, token, caCert string) (*kubernetes.Clientset, error) {
 		TLSClientConfig: rest.TLSClientConfig{CAData: decodedCert},
 	}
 
-	client, err := kubernetes.NewForConfig(config)
-	_ = client
+	client, err = kubernetes.NewForConfig(config)
```
Comment (gambol99) [inline go/api_default.go]: and probably just return as is `return kubernetes.NewForConfig(config)` and the caller will check the error

File: main.go (hunk 1 above)
Comment (danielwhatmuff) [reply main.go]: Fixed

File: go/api_default.go (hunk 4 above)
Comment (danielwhatmuff) [reply go/api_default.go]: Done

**Pull request merged**